### PR TITLE
Resolve `link list --include-topics` bug

### PR DESCRIPTION
**Pull request by sgagniere**
Release Notes
-------------

Breaking Changes
- PLACEHOLDER

New Features
- PLACEHOLDER

Bug Fixes
- Correctly display mirror topics in `confluent kafka link list --include-topics`

Checklist
---------
1. [CRUCIAL] Is the change for features that are already live in prod?
   * yes: ok

What
----
The `ListLinksResponseDataList` struct returned from `ClusterLinkingV3Api.ListKafkaMirrorTopicsUnderLinkExecute` contains an empty list of mirror topics even though a non-empty list is present in the http response itself. Until this error in either the backend or the sdk is fixed, we can restore this functionality by calling `ListKafkaMirrorTopicsUnderLink`.

This will result in extra calls, but at least it will work.

References
----------

Test & Review
-------------
Ran all tests
Manual testing with 2 mirror topics

### CODE REVIEW

File: internal/cmd/kafka/command_link_list.go [hunk 1]
```diff
@@ -83,9 +83,16 @@ func (c *linkCommand) list(cmd *cobra.Command, _ []string) error {
 
 	list := output.NewList(cmd)
 	for _, data := range listLinksRespDataList.Data {
-		if includeTopics && len(data.GetTopicsNames()) > 0 {
-			for _, topic := range data.GetTopicsNames() {
-				list.Add(newLink(data, topic))
+		if includeTopics {
+			// data.GetTopicsNames() is empty even when the http response contains a non-empty list of topic names,
```
[the remark this answers is not in the document]
Comment (sgagniere) [reply internal/cmd/kafka/command_link_list.go:87]: If it did change, it wasn't recent. I remember seeing this behavior a few months ago but I didn't recognize that it was a bug and not intended behavior (I just assumed user error on my part).

It's possible that it never worked.

File: internal/cmd/kafka/command_link_list.go [hunk 2]
```diff
@@ -104,3 +111,18 @@ func getListFields(includeTopics bool) []string {
 
 	return append(x, "SourceClusterId", "DestinationClusterId", "State", "Error", "ErrorMessage")
 }
+
+func getMirrorTopicNames(kafkaREST *pcmd.KafkaREST, clusterId, linkName string) ([]string, error) {
+	listMirrorTopicsResponseDataList, httpResp, err := kafkaREST.CloudClient.ListKafkaMirrorTopicsUnderLink(clusterId, linkName)
```
[the remark this answers is not in the document]
Comment (sgagniere) [reply internal/cmd/kafka/command_link_list.go:116]: They probably should, but this looks like another backend ticket: neither `ApiListKafkaMirrorTopicsRequest` or `ApiListKafkaLinksRequest` contain fields for page tokens or page sizes. The http response does appear to contain a next page token, but the sdk doesn't support it.

File: internal/cmd/kafka/command_link_list.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (brianstrauch) [reply internal/cmd/kafka/command_link_list.go:87]: Can you open a ticket for adding system tests for this command?

File: internal/cmd/kafka/command_link_list.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (brianstrauch) [reply internal/cmd/kafka/command_link_list.go:116]: If the API returns a next page token, then we should be able to change things on our end, right? Does the SDK not have a `PageToken()` function?

[the remark this answers is not in the document]
Comment (sgagniere) [reply internal/cmd/kafka/command_link_list.go:116]: > Does the SDK not have a PageToken() function?

Nope. Not for either of these two, and while I didn't check, I would guess it's just missing across the board in this SDK.

File: internal/cmd/kafka/command_link_list.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (sgagniere) [reply internal/cmd/kafka/command_link_list.go:87]: Yup, opened a CLI ticket for this.

File: internal/cmd/kafka/command_link_list.go [hunk 3]
```diff
@@ -104,3 +111,18 @@ func getListFields(includeTopics bool) []string {
...
+	if err != nil {
+		return nil, kafkarest.NewError(kafkaREST.CloudClient.GetUrl(), err, httpResp)
```
Comment (MuweiHe) [inline internal/cmd/kafka/command_link_list.go:118]: does this function catch the detail in http response?

Comment (sgagniere) [reply internal/cmd/kafka/command_link_list.go:118]: If the error is a `GenericOpenAPIError`, then it decodes and returns the `message` field.

File: internal/cmd/kafka/command_link_list.go [hunk 4]
```diff
@@ -104,3 +111,18 @@ func getListFields(includeTopics bool) []string {
...
+	}
+
+	mirrorTopicNames := make([]string, len(listMirrorTopicsResponseDataList.GetData()))
+
```
Comment (brianstrauch) [inline internal/cmd/kafka/command_link_list.go:122]:
```suggestion
```

**Review by brianstrauch — APPROVED** [review]
LGTM, as long as the relevant backend team has been alerted of their mistake.

**Comment by sgagniere** [comment]
@brianstrauch they elevated the priority of the ticket to 'Critical' and said they were going to ask around on how to resolve it, so they're aware!

**Comment by sgagniere** [comment]
Alright... the semaphore job is failing on `make test-installer` but it passed before the latest commit (which only deleted an empty line and so it should be unrelated), so I'm just going to go ahead and merge this in.

**Pull request merged**